### ISSUE TITLE
okta_auth_server_policy_rule  with client_credentials option produce crash 

### ISSUE THREAD

**Issue by SBerda** [okta/terraform-provider-okta#202]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.20

### Affected Resource(s)

* okta_auth_server_policy_rule

```hcl
resource "okta_auth_server_policy_rule" "my_photo_agency" {
  count = length(var.env)
  auth_server_id       = okta_auth_server.auth_server[count.index].id
  policy_id            = okta_auth_server_policy.my_photo_agency[count.index].id
  status               = "ACTIVE"
  name                 = "my_photo_agency"
  priority             = 1
  grant_type_whitelist = ["client_credentials"]
}
```

### Panic Output

https://gist.github.com/SBerda/17a7f9d150bc4dda8ac40758eba12598

### Expected Behavior

It Should create the okta_auth_server_policy_rule with the client_credential option.

### Actual Behavior

Terraform does not create the ressource and crash

### Steps to Reproduce

1. `terraform apply` with the code provided earlier.

* #0000

**Comment by bogdanprodan-okta** [comment]
Hi @SBerda ! Thanks for creating this issue! I've made an investigation and two things came up:
1) when providing `grant_type_whitelist = ["implicit"]` at least `user_whitelist` or `group_whitelist` should be present, otherwise, Okta API will return this error:
```
 Error: failed to update auth server policy rule: The API returned an error: Api validation failed: conditions.people. Causes:
 errorSummary: conditions.people: At least one of ''conditions.people.users.include'' or ''conditions.people.groups.include''
 must contain a valid user or group when ''conditions.grantTypes'' contains one or more of [authorization_code, implicit,
 password].
```
2) there is a bug in the code that leads to this panic. I'll fix that in the new v3.7.2 version.

**Comment by bogdanprodan-okta** [comment]
Hi @SBerda! The issue should be fixed in the latest 3.7.2 version. In case it still exists, please reopen this issue.
